crates/sui-graphql-rpc/src/types/move_type.rs

```diff
@@ -101,19 +101,17 @@ impl MoveType {
     /// Structured representation of the type signature.
     async fn signature(&self) -> Result<MoveTypeSignature> {
         // Factor out into its own non-GraphQL, non-async function for better testability
-        self.signature_impl()
+        self.signature_impl().extend()
     }
 
     /// Structured representation of the "shape" of values that match this type.
     async fn layout(&self, ctx: &Context<'_>) -> Result<MoveTypeLayout> {
-        let resolver: &Resolver<PackageCache> = ctx.data().map_err(|_| {
-            graphql_error(
-                code::INTERNAL_SERVER_ERROR,
-                "Unable to fetch Package Cache.",
-            )
-        })?;
-
-        MoveTypeLayout::try_from(self.layout_impl(resolver).await?)
+        let resolver: &Resolver<PackageCache> = ctx
+            .data()
+            .map_err(|_| Error::Internal("Unable to fetch Package Cache.".to_string()))
+            .extend()?;
+
+        MoveTypeLayout::try_from(self.layout_impl(resolver).await.extend()?).extend()
```

Comment:
damn, good catch

Reply:
I'm pretty sure I also made this mistake ;)